java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/BuildPipelineConfigurator.java

```diff
@@ -108,6 +117,8 @@ private BuildContext configureBuildContext(Path workspaceDir) throws IOException
         ? parseAppYaml(pathToAppYaml.get())
         : new AppYaml();
 
+    appYaml.applyOverrideSettings(this.overrideSettings);
```

Comment:
It looks like override settings were never actually applied previously without this line.
Currently, if I remove this line all tests still pass. It doesn't seem right.

Reply:
Yes previously this line was done inside the parser, which would work if the app.yaml file contained only part of the required settings, but wouldn't run at all if no app.yaml file was given or found at all.

So now it is moved out here. I'll add the unit tests for this too.